### PR TITLE
megadock: new package at 4.0.3

### PR DESCRIPTION
**Pull request by codeandkey**

### CODE REVIEW

File: var/spack/repos/builtin/packages/megadock/package.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        # set mpi, cuda installations
+        if '+mpi' in spec:
+            env['MPICOMPILER'] = self.spec['mpi'].mpicxx
+
+        if '+cuda' in spec:
+            env['CUDA_INSTALL_PATH'] = self.spec['cuda'].prefix
+
+        env['FFTW_INSTALL_PATH'] = self.spec['fftw'].prefix
+        env['CPPCOMPILER'] = 'c++'
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/megadock/package.py]: Setting environment variables should be done in the `setup_environment` method. For cases like this one you should modify `spack_env` (the first argument to the function). As an example implementation:

https://github.com/spack/spack/blob/d4d402586c949b167334c7f53f2889017ae6ce9d/var/spack/repos/builtin/packages/go/package.py#L97-L102

File: var/spack/repos/builtin/packages/megadock/package.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        # use -fopenmp flag if gcc
+        if '%gcc' in spec:
+            env['OMPFLAG'] = '-fopenmp'
```
[the remark this answers is not in the document]
Comment (molecuul) [reply var/spack/repos/builtin/packages/megadock/package.py]: The package uses ‘-openmp’ by default (for intel), this overrides it when gcc is used.

[the remark this answers is not in the document]
Comment (alalazo) [reply var/spack/repos/builtin/packages/megadock/package.py]: In this case I think it's better to employ `self.compiler.openmp_flag` unconditionally.

**Comment by molecuul** [comment]
Thanks for the feedback. I think I've addressed most of the issues.
libimf is now preferred on intel but I don't know if this needs to be expanded to cover all of spack's supported compilers.

File: var/spack/repos/builtin/packages/megadock/package.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        filter_file('/opt/cuda/6.5/samples', '$(CUDA_INSTALL_PATH)/samples',
+                    'Makefile', string=True)
+
+        # need to link calcrg to compiler's math impl
+        # libm seems to be present in most compilers
+        mathlib = '-lm'
+
+        # prefer libimf with intel
+        if '%intel' in spec:
+            mathlib = '-limf'
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/megadock/package.py]: FYI, the few lines above can be written:
```python
mathlib = '-limf' if '%intel' in spec else '-lm'
```
to make the package more compact / readable.

**Pull request merged**